x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/index.tsx

```diff
@@ -0,0 +1,290 @@
+   * returns a `ReactWrapper` even if nothing is found, as that is how `enzyme` does things.
+   */
+  public processNodeElements(options: ProcessNodeElementSelectorOptions = {}): ReactWrapper {
+    return this.findInDOM(processNodeElementSelector(options));
+  }
+
+  /**
+   * true if a process node element is found for the entityID and if it has an [aria-selected] attribute.
+   */
+  public processNodeElementLooksSelected(entityID: string): boolean {
+    return this.processNodeElements({ entityID, selected: true }).length === 1;
+  }
+
+  /**
+   * true if a process node element is found for the entityID and if it *does not have* an [aria-selected] attribute.
+   */
+  public processNodeElementLooksUnselected(entityID: string): boolean {
+    //  find the process node, then exclude it if its selected.
+    return (
+      this.processNodeElements({ entityID }).not(
+        processNodeElementSelector({ entityID, selected: true })
+      ).length === 1
+    );
+  }
+
+  /**
+   * Return the selected node query string values.
+   */
+  public queryStringValues(): { selectedNode: string[] } {
+    const urlSearchParams = new URLSearchParams(this.history.location.search);
+    return {
+      selectedNode: urlSearchParams.getAll(`resolver-${this.resolverComponentInstanceID}-id`),
+    };
+  }
+
+  /**
+   * The element that shows when Resolver is waiting for the graph data.
+   */
+  public graphLoadingElement(): ReactWrapper {
+    return this.findInDOM('[data-test-subj="resolver:graph:loading"]');
+  }
+
+  /**
+   * The element that shows if Resolver couldn't draw the graph.
+   */
+  public graphErrorElement(): ReactWrapper {
+    return this.findInDOM('[data-test-subj="resolver:graph:error"]');
+  }
+
+  /**
+   * The element where nodes get drawn.
+   */
+  public graphElement(): ReactWrapper {
+    return this.findInDOM('[data-test-subj="resolver:graph"]');
+  }
+
+  /**
+   * Like `this.wrapper.find` but only returns DOM nodes.
+   */
+  private findInDOM(selector: string): ReactWrapper {
+    return this.wrapper.find(selector).filterWhere((wrapper) => typeof wrapper.type() === 'string');
+  }
+}
+
+const baseResolverSelector = '[data-test-subj="resolver:node"]';
+
+interface ProcessNodeElementSelectorOptions {
+  /**
+   * Entity ID of the node. If passed, will be used to create an data-attribtue CSS selector that should only get the related node element.
+   */
+  entityID?: string;
+  /**
+   * If true, only get nodes with an `[aria-selected="true"]` attribute.
+   */
+  selected?: boolean;
+}
+
+/**
+ * An `enzyme` supported CSS selector for process node elements.
+ */
+function processNodeElementSelector({
+  entityID,
+  selected = false,
+}: ProcessNodeElementSelectorOptions = {}): string {
+  let selector: string = baseResolverSelector;
+  if (entityID !== undefined) {
+    selector += `[data-test-resolver-node-id="${entityID}"]`;
+  }
+  if (selected) {
+    selector += '[aria-selected="true"]';
```

Comment:
:grey_question: Seems like this would break if more than one thing was `aria-selected` wouldn't it?

Reply:
it gets added to `baseResolverSelector` which is `'[data-test-subj="resolver:node"]'` which is intended to be unique ish. although that'll break as well w/ more than 1 resolver on the page. we should probably add the document location ID to our data-test-subjs

Reply:
@bkimmel, when would we have more than one process node selected?

Reply:
if we did have more than 1 process selected in a single resolver, this would return both, which would be what the test code would want (so we can say 'expected 1, got 2') or whatever

Reply:
@michaelolo24 I was just thinking multi-resolver, mainly